### PR TITLE
docs-website for github integration

### PR DESCRIPTION
**Pull request by codingrealmdinesh**
docswebsite

### CODE REVIEW

**Comment by ally-sassman** [comment]
@ubanerjeeNR This is still in draft mode, but I want to put it on your radar as the OHI liason.

**Comment by pshadowrealm** [comment]
@ubanerjeeNR please review the PR once free.

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+title: Github integration
+tags:
+  - New Relic integrations
+  - Github integration
+metaDescription: Set up automatic ingestion of your GitHub Dependabot events by using webhook for ongoing ingestion and historical import for past data.
+freshnessValidatedDate: 2024-10-29
+---
+
+
+Set up automatic ingestion of your GitHub Dependabot events by using webhook for ongoing ingestion and historical import for past data.
```
Comment (ubanerjeeNR) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:11]: Can we please make the problem statement clearer in the description? What problem does this integration solve and what's in it for the users should come through clearly. For example, does the following work?

The title could be: Enhance security monitoring by integrating GitHub Dependabot with New Relic

The description could be:

Streamline vulnerability management and dependency tracking by automatically ingesting GitHub Dependabot events into New Relic. To ingest real-time events to New Relic, set up a webhook. To import past data, set up a historical import mechanism. This integration helps you optimize dependency management and reduce risk.

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  alt="Github Integration dashboard"
+  src="/images/github-integration-screenshot.webp"
+/>
+
+<figcaption>
+After setting up our Github integration, see your data of teams, repos and users right out of the box.
+</figcaption>
+
+
+## Set up the Github integration [#set-up-github]
```
Comment (ubanerjeeNR) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:24]: Before we provide procedure to perform the actual setup, don't we need to call out prerequisites for the setup? What conditions must a user meet before the user can proceed with the setup? If there are none, please ignore.

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+Complete the following steps to set up the Github integration:
+
+<Steps>
+<Step>
+
+## Integration Details [#integration-details]
+
+By default, the Power Service catalog is selected, which will ensure to import all the data from Teams, Repositories and Users.
```
Comment (ubanerjeeNR) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:33]: This lacks context. Where am I performing this? Also, as per the instruction, I don't really have to perform anything. It says Power Service catalog is the default selection. And if that's true indeed, then do we really need a step. All we need is to call out that the catalog must be selected for the setup.

Also, what's a Power Service catalog? Don't we need to explain that? Can we assume users already know this? If latter, then that should get called out as a prereq.

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    </Step>
+
+    <Step>
+
+## Github Integration [#github-integration]
+
+1. In this page, once we click on the install button, we get redirected to the Github App installation page. From there we need to install the Github app into the organization.
```
Comment (ubanerjeeNR) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:40]: Missing context: which page are we referring to here? Where are we performing these steps? Which GitHub app are we installing? Into which organization?

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+2. If we are successfully able to install, it will show a success message with the teams, repo and users lists or else a failure message.
```
Comment (ubanerjeeNR) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:41]: A step is an action that a user performs. This is more like the possible outcome of a user action and therefore shouldn't be numbered.

File: src/nav/infrastructure.yml [hunk 6]
```diff
@@ -168,6 +168,8 @@ pages:
             path: /docs/infrastructure/host-integrations/host-integrations-list/envoy-integration
           - title: F5 integration
             path: /install/f5
+          - title: Github Integration
+            path: /docs/infrastructure/host-integrations/host-integrations-list/
```
Comment (ubanerjeeNR) [inline src/nav/infrastructure.yml:172]: I am unable to open this page by clicking the corresponding entry in the left nav. Not sure why though. I notice there's a forward slash at the end of the path. Can we please delete it and check if that does the trick?

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx (hunk 5 above)
Comment (ubanerjeeNR) [reply src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:41]: What happens after the setup? What's next for the user? Don't we need to show what kind of data is ingested in New Relic and how to make sense of that data? Take a look at this page: https://deploy-preview-19113--docs-website-netlify.netlify.app/docs/infrastructure/host-integrations/host-integrations-list/active-directory-integration/, for example. It lists all the metrics, events, and logs that one sees in New Relic as a result of the integration. Without that, this information would be incomplete.

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx (hunk 3 above)
Comment (pshadowrealm) [reply src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:33]: The selection is by default for our MVP. But latter on we are planning to make that optional. So broadly speaking its not by default . I will update the text.

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx (hunk 2 above)
Comment (pshadowrealm) [reply src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:28]: Added prerequisites.

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx (hunk 1 above)
Comment (pshadowrealm) [reply src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx]: Updated!

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx (hunk 4 above)
Comment (pshadowrealm) [reply src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx]: Added context!

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx (hunk 5 above)
Comment (pshadowrealm) [reply src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx]: Added result page content!

File: src/nav/infrastructure.yml (hunk 6 above)
Comment (pshadowrealm) [reply src/nav/infrastructure.yml]: Can you open now? Please check and let me know.

**Comment by pshadowrealm** [comment]
All the comments are resolved IMO, Please check @ubanerjeeNR .

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx [hunk 7]
```diff
@@ -0,0 +1,64 @@
+---
+title: Automating New Relic Service Catalog onboarding with GitHub Integration to enhance user adoption
```
Comment (ubanerjeeNR) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:2]: Seamlessly adopt New Relic Service catalog with the GitHub integration

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+tags:
+  - New Relic integrations
+  - Github integration
+metaDescription: Set up automatic ingestion of your GitHub Dependabot events by using webhook for ongoing ingestion.
+freshnessValidatedDate: 2024-10-29
+---
+
+
+Establishing a New Relic Service Catalog offers immense value but currently demands significant time and resource investments from users. The process involves detailed structuring of teams, user creation, assignments, and mapping of entity ownership, presenting a substantial barrier to entry. To mitigate this, we propose an automation solution that leverages data from external systems, such as GitHub, where team structures and associations are already managed. This integration aims to streamline the setup process, significantly reduce upfront efforts, and accelerate the adoption of New Relic's Service Catalog.
```
Comment (ubanerjeeNR) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:11]: Thanks for providing this context. This is great. I have made minor tweaks.

Implementing a New Relic Service Catalog can unlock considerable benefits, but at present, it requires a considerable investment of time and resources. You have to take care of team orchestration, user provisioning, role assignments, and entity ownership delineation, which create a significant threshold for entry. To alleviate this challenge, we suggest the adoption of an automation solution that capitalizes on existing team configurations and relationships maintained in external systems such as GitHub. This simplifies the initial configuration, dramatically lessen the required effort, and accelerates the adoption of the New Relic Service Catalog, making its powerful insights and organization tools more accessible to users.

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  src="/images/github-integration-screenshot.webp"
+/>
+
+<figcaption>
+After setting up our Github integration, see your data of teams, repos and users right out of the box.
+</figcaption>
+
+## Prerequisites [#prerequisites]
+
+Users need to ensure that they are logged in to their Github accounts. This needs to be the same github account in which they want to install the github app that New Relic will be providing during installation step.
```
Comment (ubanerjeeNR) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:25]: GitHub is always GitHub. And never Github or github. The casing is inconsistent across the topic. Can we please correct this?

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+
+## Set up the Github integration [#set-up-github]
+
+Vist <DNT>**[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > + Integrations & Agents** > Github integration</DNT> tile and follow the below steps to complete the Github integration.
+
+<Steps>
+<Step>
+
+## Service Catalog Details [#service-catalog-details]
```
Comment (ubanerjeeNR) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/github-integration.mdx:35]: Can we please provide a couple of relevant screenshots for visual reference?

**Comment by dinesh-newrelic** [comment]
Need to close this since I havent used NR account to create the pr.

**Comment by ubanerjeeNR** [comment]
This pr was created using a non-NR account. Hence closing this. @dinesh-newrelic will create a separate PR with the NR account to handle these changes.

**Pull request closed without merging**